src/pudl_archiver/__init__.py

```diff
@@ -79,3 +81,13 @@ async def archive_datasets(
                 f"Encountered exceptions, showing traceback for last one: {[repr(e) for e in exceptions]}"
             )
             raise exceptions[-1][1]
+
+    if summary_file is not None:
+        run_summaries = [result.dict() for _, result in results]
+
+        with open(summary_file, "w") as f:
+            json.dump(run_summaries, f, indent=2)
+
+    validation_results = [result.success for _, result in results]
+    if not all(validation_results):
+        raise RuntimeError("Error: archive validation tests failed.")
```

Comment:
Is this your imagined workflow?:

1. see this
2. shoot, need to go look at the run summary file

Reply:
Yeah I thought about printing out any failed validation tests here, but decided it was sufficient to just go to the file if there's a failure. I'm definitely open to changing this though